.github/workflows/ci-cd.yml

```
@@ -57,60 +88,32 @@ jobs:
       if: matrix.python-version == '3.11'
       uses: codecov/codecov-action@v4.5.0
       with:
-        token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+        token: ${{ secrets.CODECOV_TOKEN }}  # not required for public repos
         files: ./coverage.xml
-        flags: unittests # optional
-        name: codecov-umbrella # optional
-        fail_ci_if_error: true # optional (default = false)
-        verbose: true # optional (default = false)
-
-
-  pre-deploy:
-    name: Pre-Deploy
-    runs-on: ubuntu-latest
-    needs: test
-    # Run only on pushing a tag
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
-    steps:
-      - name: Dummy
-        run: |
-            echo "Predeploy step"
+        flags: unittests  # optional
+        name: codecov-umbrella  # optional
+        fail_ci_if_error: true  # optional (default = false)
+        verbose: true  # optional (default = false)
 
-  build-tarball:
-    name: Tarball
-    runs-on: ubuntu-latest
-    needs: pre-deploy
-    steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-      with:
-        submodules: true
-    - name: Setup Python 3.8
-      uses: actions/setup-python@v5
-      with:
-        python-version: 3.8
-    - name: Make distributions (sdist and wheel)
-      run: |
-        python -c "import sys; print(f'Python version: {sys.version}')"
-        python -m pip install -U setuptools pip wheel
-        python setup.py sdist bdist_wheel
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v4
-      with:
-        name: dist
-        path: dist
-
-  deploy:
-    name: Deploy
-    needs: [build-tarball]
+  pypi-publish:
+    name: Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish on tag pushes
```

This check is the old way, I think GH implemented a helper variable after I added this to the guide. I should probably update the examples :)

```suggestion
    if: github.ref_type == 'tag'
```